packages/terra-form-select/src/single/Frame.jsx

```diff
@@ -382,18 +382,26 @@ class Frame extends React.Component {
     return ariaLabel === undefined ? defaultAriaLabel : ariaLabel;
   }
 
+  /**
+   * Generates the main label for the field.
+   */
+  mainLabel() {
+    let label;
+    if (this.props.display) {
+      label = `${this.props.display} is selected, ${this.ariaLabel()}`;
+    } else if (this.props.placeholder) {
+      label = `${this.props.placeholder}, ${this.ariaLabel()}`;
+    }
+
+    return label || this.ariaLabel();
+  }
+
   /**
    * Determines compatible role attribute to apply to select based on active variant and disabled prop
    */
   role() {
-    let role;
     const { disabled } = this.props;
-
-    if (SharedUtil.isMac()) {
-      role = SharedUtil.isSafari() ? 'group' : 'button';
-    } else {
-      role = 'combobox';
-    }
+    const role = SharedUtil.isSafari() ? 'group' : 'button';
```

Comment:
Does safari fails to identify any of the workflows if role `button` is used ..??

Reply:
When "role=button" is used, screen reader has issues in announcing the list items. Due to this, had to retain "aria-live=polite" along with use of "role=button" for safari only.